pydantic_settings/sources.py

```diff
@@ -1912,7 +1912,10 @@ def _read_files(self, files: PathType | None) -> dict[str, Any]:
         for file in files:
             file_path = Path(file).expanduser()
             if file_path.is_file():
-                vars.update(self._read_file(file_path))
+                try:
+                    vars.update(self._read_file(file_path))
+                except ValueError as e:
+                    raise SettingsError(f'Parsing error encountered for {file_path}: {e}')
```

Comment:
I think we could improve the user experience by explicitly checking that the result of file read is a dict because otherwise, (e.g. in added tests), the `ValueError`, *which is about `dict.update()`*, is not very informative:
```python
>>> d
{}
>>> d.update("abc")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: dictionary update sequence element #0 has length 1; 2 is required
```

Also, I realize that calling `dict.update()` with an iterable is also valid and so loading a YAML like:
```yaml
- - x
  - y
- - a
  - b
```
would work; not sure if it's expected? But maybe it's another issue.

Reply:
we can probably check if the object returned by `self._read_file` is a dict or not and handle that specific case.

There are also a lot of cases where reading from (toml, yaml,...) file will raise ValueError.  Eg:

``` python
 import tomli

tomli.loads("demo =\ndemo2 =")
```

Will throw a tomli.TOMLDecodeError :/ finally i think we should simply keep the `ValueError`.